core/trino-main/src/main/java/io/trino/execution/scheduler/BinPackingNodeAllocatorService.java

```diff
@@ -378,6 +379,16 @@ public void release()
                 throw new IllegalStateException("Node " + node + " already released");
             }
         }
+
+        public void deallocateMemory(InternalNode node)
+        {
+            // guarding deallocation on atomic boolean.
+            // Otherwise, memory could be deallocated twice in case of race between
+            // call to BinPackingNodeLease.release() and deallocation in processPendingAcquires().
```

Comment:
I'd go without comment

- "guarding deallocation on atomic boolean" -- this is exactly what the code is doing, obvious 
- "Otherwise, memory could be deallocated twice" -- well, this is part is also obvious
- "BinPackingNodeLease.release() and deallocation in processPendingAcquires()." -- redundant, the callers are easily findable via "Call Hierarchy".